src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -372,19 +383,67 @@ where
 
             self.state.collections.insert(id, collection_state);
 
-            dataflow_commands.push(CreateSourceCommand {
-                id,
-                desc,
-                since,
-                storage_metadata: self.collection_metadata(id)?,
+            let storage_service = self
+                .orchestrator
+                .ensure_service(
+                    &id.to_string(),
+                    ServiceConfig {
+                        image: self.storaged_image.clone(),
+                        args: &|assigned| {
+                            vec![
+                                format!("--workers=1"),
+                                format!(
+                                    "--listen-addr={}:{}",
+                                    assigned.listen_host, assigned.ports["controller"]
+                                ),
+                                format!(
+                                    "--http-console-addr={}:{}",
+                                    assigned.listen_host, assigned.ports["http"]
+                                ),
+                            ]
+                        },
+                        ports: vec![
+                            ServicePort {
+                                name: "controller".into(),
+                                port_hint: 2100,
+                            },
+                            ServicePort {
+                                name: "http".into(),
+                                port_hint: 6875,
+                            },
+                        ],
+                        // TODO: limits?
+                        cpu_limit: None,
+                        memory_limit: None,
+                        scale: NonZeroUsize::new(1).unwrap(),
+                        labels: HashMap::new(),
+                        availability_zone: None,
```

Comment:
No need to block this for a prototype but something we should be careful about, if we care about co-locating `storaged` pods in an AZ for network traffic reasons (ingress / egress, latency)

Reply:
Yeah, we'll need to think through how we want to allow users to constrain their sources to AZs!